File: Games/GameOfGames/src/main/java/jokatu/game/games/gameofgames/input/CreateGameInputDeserialiser.java
package jokatu.game.games.gameofgames.input;

import jokatu.game.input.DeserialisationException;
import jokatu.game.input.TypedSingleKeyInputDeserialiser;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

import java.util.Map;

@Component
public class CreateGameInputDeserialiser extends TypedSingleKeyInputDeserialiser<String, CreateGameInput> {
	@NotNull
	@Override
	public CreateGameInput deserialiseTypedSingleValue(@NotNull Map<String, Object> json, @NotNull String gameName) throws DeserialisationException {
		return new CreateGameInput(gameName);
	}

	@NotNull
	@Override
	protected Class<String> getType() {
		return String.class;
	}

	@NotNull
	@Override
	protected String getKeyName() {
		return "gameName";
	}
}

File: Games/GameOfGames/src/main/kotlin/jokatu/game/games/gameofgames/input/CreateGameInputDeserialiser.kt
package jokatu.game.games.gameofgames.input

import jokatu.game.input.DeserialisationException
import jokatu.game.input.TypedSingleKeyInputDeserialiser
import org.springframework.stereotype.Component

@Component
class CreateGameInputDeserialiser : TypedSingleKeyInputDeserialiser<String, CreateGameInput>(String::class) {

	override val keyName: String
		get() = "gameName"

	@Throws(DeserialisationException::class)
	override fun deserialiseTypedSingleValue(json: Map<String, Any>, gameName: String): CreateGameInput {
		return CreateGameInput(gameName)
	}
}
